db/system_keyspace.cc

```diff
@@ -52,6 +52,21 @@
 
 using days = std::chrono::duration<int, std::ratio<24 * 3600>>;
 
+template <>
+inline sstables::basic_info&& value_cast<sstables::basic_info>(data_value&& value) {
+    auto* user_type = dynamic_cast<const user_type_impl*>(&value._type->without_reversed());
+    auto* field_values = (const user_type_impl::native_type*)value._value;
+
+    static thread_local sstables::basic_info native_value; // Static so it survives return from this function.
```

Comment:
why does it need to survive return?

Reply:
Ah, I see you return `rvalue` ref, but why?

Reply:
Due to https://github.com/paszkow/scylladb/blob/issue-3791/types/types.hh#L513. See the specialization for `timeuuid_native_type` a few lines below as well.

Reply:
But then this is not a cast anymore. It is a conversion.

Reply:
Dropped.